File: Code Chef/C Programming/800-899/841 Which Mixture MIXTURE.c
#include <stdio.h>
int main(void)
{
	int t,a,b;
	scanf("%d",&t);
	for(int i=0;i<t;i++)
	{
	    scanf("%d %d",&a,&b);
	    if(a==0)
	    {
	        printf("Liquid\n");
	    }
	    else if(b==0)
	    {
	        printf("Solid\n");
	    }
	    else if((a>0)&&(b>0))
	    {
	        printf("Solution\n");
	    }
	}
	return 0;
}

File: Code Chef/JavaScript/800-899/841 Which Mixture MIXTURE.js
let input = '';
process.stdin.on('data', (data) => input += data);

const strToInt = (strArray) => {
    strArray.shift();
    const intArray = strArray.map((strItem) => strItem.trim().split(' ').map(Number));
    
    return intArray;
};

const classifyMixtureType = (unitsOfItemA, unitsOfItemB) => {
    const isSolution = (unitsOfItemA > 0) && (unitsOfItemB > 0);
    const isLiquid = (unitsOfItemA === 0) && (unitsOfItemB !== 0);
    
    const mixtureType = (isSolution) ? 'Solution'
        : (isLiquid) ? 'Liquid'
        : 'Solid'
    ;
    console.log(mixtureType);
};

process.stdin.on('end', () => {
    const testcases = strToInt(input.trim().split('\n'));
    
    testcases.forEach((inputs) => {
       const [unitsOfItemA, unitsOfItemB] = inputs; 
       classifyMixtureType(unitsOfItemA, unitsOfItemB);
    });
});
